scapy/contrib/gtp_v2.py

```diff
@@ -657,26 +734,41 @@ class IE_MSISDN(gtp.IE_Base):
 class IE_Indication(gtp.IE_Base):
     name = "IE Indication"
     fields_desc = [ByteEnumField("ietype", 77, IEType),
-                   ShortField("length", None),
+                   FieldLenField("length", 8),
```

Comment:
Looks good except this field; not sure this works?

Reply:
Yes, worked well so far. Since it has all conditional IEs, if user does not want to set any value, it will be encoded as 0 otherwise 1.
Hence total length of this IE will always be 8.

Reply:
In that case don't use `FieldLenField` ? Since it's not hooked to any fields..

Reply:
okay, I will make it as ShortField

Reply:
So, 3gpp specs are little confusing for me:
{code}
For each message the applicable flags of the Indication IE shall be clearly specified in the individual message sub clause. The remaining flags of the Indication IE not so indicated shall be discarded by the receiver.
The receiver shall consider the value of the applicable flags as "0", if the Indication IE is applicable for the message but not included in the message by the sender. 
{code}
Thus, to be on the safer side, I modified it to compute the length automatically based on total length of IE. Verified same:
{code}
>>> ie = IE_Indication(ietype='Indication', PS=1, CPRAI=1)
>>> ie
<IE_Indication  ietype=Indication PS=1 CPRAI=1 |>
>>> ie.PPON 
0
>>> sendp(Ether()/IP(src='10.15.15.61', dst='10.9.221.58')/UDP(sport=2123, dport=2123)/GTPHeader(seq=1, P=0, T=1, gtp_type=36, teid=1, length=20)/GTPV2CreateSessionResponse(IE_list=[ie]), iface='eth0')
.
Sent 1 packets.
>>> a = 'd89ef3da40e2fa163e956dce0800450000340001000040117a290a0f0f3d0a09dd3a084b084b00204fc74824001400000001000001004d0008000008008000000000'
>>> gtp = Ether(hex_bytes(a))
>>> gtp.IE_list[0]
<IE_Indication  ietype=Indication length=8 CR_flag=0 instance=0 DAF=0 DTF=0 HI=0 DFI=0 OI=0 ISRSI=0 ISRAI=0 SGWCI=0 SQCI=0 UIMSI=0 CFSI=0 CRSI=0 PS=1 PT=0 SI=0 MSV=0 RetLoc=0 PBIC=0 SRNI=0 S6AF=0 S4AF=0 MBMDT=0 ISRAU=0 CCRSI=0 CPRAI=1 ARRL=0 PPOFF=0 PPON=0 PPSI=0 CSFBI=0 CLII=0 CPSR=0 NSI=0 UASI=0 DTCI=0 BDWI=0 PSCI=0 PCRI=0 AOSI=0 AOPI=0 ROAAI=0 EPCOSI=0 CPOPCI=0 PMTSMI=0 S11TF=0 PNSI=0 UNACCSI=0 WPMSI=0 5GSNN26=0 REPREFI=0 5GSIWKI=0 EEVRSI=0 LTEMUI=0 LTEMPI=0 ENBCRSI=0 TSPCMI=0 Spare=0 Spare=0 Spare=0 N5GNMI=0 5GCNRS=0 5GCNRI=0 5SRHOI=0 ETHPDN=0 |>
>>> gtp.IE_list[0].length
8
{code}